lib/softoken/pkcs11c.c

```diff
@@ -1985,6 +1986,67 @@ sftk_doHMACInit(SFTKSessionContext *context, HASH_HashType hash,
     return CKR_OK;
 }
 
+/*
+ * common CMAC initialization routine
+ */
+static CK_RV
+sftk_doCMACInit(SFTKSessionContext *context, CMACCiphers type,
+                SFTKObject *key, CK_ULONG mac_size)
+{
+    SFTKAttribute *keyval;
+    CMACContext *CMACcontext;
+    CK_ULONG *intpointer;
+
+    /* TODO: check key size: need FIPS recommendation */
```

Comment:
I'll remove this comment prior to adding to Mercurial. I realized that the key size is static based on the underlying block cipher. `AES_CreateContext` will reject invalid key sizes. Never mind!

Reply:
Ok, sounds good.